### PR TITLE
add getstorage RPC call

### PR DESCRIPTION
**Pull request by ixje**
**What current issue(s) does this address?, or what feature is it adding?**
add `getstorage` rpc call (thanks to @localhuman for spotting the reversed hash requirement)

**How did you solve this problem?**
-

**How did you make sure your solution works?**
tests

**Did you add any tests?**
yes

**Are there any special changes in the code that we should be aware of?**
no

**Did you run `make lint` and `make test`?**
yes
**Are you making a PR to a feature branch or development rather than master?**
feature

### CODE REVIEW

File: neo/api/JSONRPC/JsonRpcApi.py [hunk 1]
```diff
@@ -216,7 +224,20 @@ def json_rpc_method_handler(self, method, params):
             raise NotImplementedError()
 
         elif method == "getstorage":
-            raise NotImplementedError()
+            reverse_hash = bytearray.fromhex(params[0])
+            reverse_hash.reverse()
+            script_hash = UInt160(data=reverse_hash)
+
+            if len(params) < 2:  # note this is not in the original RPC server,I assume they forgot this as it throws an exception
+                raise JsonRpcError(-100, "Invalid key")
```
Comment (metachris) [inline neo/api/JSONRPC/JsonRpcApi.py]: better error would be "missing second argument 'key'"

File: neo/api/JSONRPC/JsonRpcApi.py [hunk 2]
```diff
@@ -216,7 +224,20 @@ def json_rpc_method_handler(self, method, params):
             raise NotImplementedError()
 
         elif method == "getstorage":
-            raise NotImplementedError()
+            reverse_hash = bytearray.fromhex(params[0])
```
Comment (metachris) [inline neo/api/JSONRPC/JsonRpcApi.py]: quite a few lines of code, i'd prefer to move this into it's own method

**Comment by metachris** [comment]
👍 

Looks good to me

**Comment by localhuman** [comment]
This looks good, but I had already added in my branch with some other changes in it.

**Comment by ixje** [comment]
ah didn't know. Will close this then

**Pull request closed without merging**